src/promptflow/promptflow/azure/operations/_flow_operations.py

```diff
@@ -54,6 +71,182 @@ def __init__(
         self._service_caller = service_caller
         self._credential = credential
 
+    @cached_property
+    def _common_azure_url_pattern(self):
+        operation_scope = self._operation_scope
+        url = (
+            f"/subscriptions/{operation_scope.subscription_id}"
+            f"/resourceGroups/{operation_scope.resource_group_name}"
+            f"/providers/Microsoft.MachineLearningServices"
+            f"/workspaces/{operation_scope.workspace_name}"
+        )
+        return url
+
+    def _get_flow_portal_url(self, flow_resource_id: str):
+        """Get the portal url for the run."""
+        match = self._FLOW_RESOURCE_PATTERN.match(flow_resource_id)
+        if not match or len(match.groups()) != 2:
+            logger.warning("Failed to parse flow resource id '%s'", flow_resource_id)
+            return None
+        datastore, flow_id = match.groups()
+        url = f"https://ml.azure.com/prompts/flow/{datastore}/{flow_id}/details?wsid={self._common_azure_url_pattern}"
```

Comment:
Nit: other clouds & VNet may have different endpoint, e.g.: "https://ml.azure.cn/" for azure china cloud. Maybe we could check if we can ask service side return this instead of client side hard code.

Reply:
Good point, will confirm with server side in next pr.